simple/rdm_atomic.c

```diff
@@ -773,32 +741,45 @@ int main(int argc, char **argv)
 				run_all_ops = 0;
 				op_type = get_fi_op(optarg);
 				if (op_type == FI_ATOMIC_OP_LAST)
-					usage(argv[0]);
-			}
-			break;
-		case 'S':
-			if (!strncasecmp("all", optarg, 3)) {
-				size_option = 1;
-				run_all_sizes = 1;
-			} else {
-				run_all_sizes = 0;
-				transfer_size = atoi(optarg);
+					ft_csusage(argv[0], NULL);
 			}
 			break;
+		case 'v':
+			ft_version(argv[0]);
+			return EXIT_SUCCESS;
 		default:
-			usage(argv[0]);
-			return -1;
+			ft_parseinfo(op, optarg, &hints);
+			ft_parsecsopts(op, optarg, &opts);
+			break;
+		case '?':
+		case 'h':
+			ft_csusage(argv[0], "Ping pong client and server using atomic ops.");
+			fprintf(stderr, "  -o <op>\tatomic op type: all|min|max|read|write|cswap (default: all)]\n");
+			return EXIT_FAILURE;
+		}
+	}
+
+	if (optind < argc)
+		opts.dst_addr = argv[optind];
+
+	if (opts.src_addr) {
+		ret = ft_getsrcaddr(opts.src_addr, opts.port, &hints);
+
+		if (ret) {
+			FI_DEBUG("source address error %s\n", gai_strerror(ret));
+			return EXIT_FAILURE;
 		}
 	}
 
-	hints.domain_attr = &domain_hints;
-	hints.ep_attr = &ep_hints;
 	hints.ep_type = FI_EP_RDM;
 	hints.caps = FI_MSG | FI_ATOMICS | FI_BUFFERED_RECV;
 	hints.mode = FI_CONTEXT | FI_LOCAL_MR | FI_PROV_MR_ATTR;
 	hints.addr_format = FI_FORMAT_UNSPEC;
-	
-	ret = run();
 
-	return ret;
+	if (opts.prhints) {
+		printf("%s", fi_tostr(&hints, FI_TYPE_INFO));
+		return EXIT_SUCCESS;
+	}
```

Comment:
Gotcha -- I have the same setup in my other test that I'm developing.  That will be a superset of these tests, so I don't think this will be needed.